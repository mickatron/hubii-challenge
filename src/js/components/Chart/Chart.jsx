import classnames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import { scaleTime } from 'd3-scale';
import { utcDay } from 'd3-time';

import { ChartCanvas, Chart } from 'react-stockcharts';
import { CandlestickSeries } from 'react-stockcharts/lib/series';
import { XAxis, YAxis } from 'react-stockcharts/lib/axes';
import { last, timeIntervalBarWidth } from 'react-stockcharts/lib/utils';

import classNameType from 'js/propTypes/className';

export default class CandleStickChart extends React.Component {
  static propTypes = {
    className: classNameType,
    data: PropTypes.arrayOf(PropTypes.shape({
      close: PropTypes.number.isRequired, // 25.710416
      date: PropTypes.string.isRequired, // "Mon Jan 04 2010 00:00:00 GMT+1100...."
      high: PropTypes.number.isRequired, // 25.835021381744056
      low: PropTypes.number.isRequired, // 25.411360259406774
      open: PropTypes.number.isRequired, // 25.436282332605284
      volume: PropTypes.number.isRequired, // 38409100
    })).isRequired,
    ratio: PropTypes.number,
    width: PropTypes.number,
  };

  static defaultProps = {
    className: undefined,
    ratio: undefined,
    width: undefined,
  };

  render() {
    const {
      className,
      data,
      ratio,
      width,
    } = this.props;
    const xAccessor = d => d.date;
    const xExtents = [
      xAccessor(last(data)),
      xAccessor(-data[data.length]),
    ];

    return (
      <ChartCanvas
        className={classnames(className, 'Chart')}
        height={400}
        ratio={ratio}
        width={width}
        margin={{
          left: 50, right: 50, top: 10, bottom: 30,
        }}
        type="svg"
        seriesName="HubiiChallenge"
        data={data}
        displayXAccessor={xAccessor}
        xScale={scaleTime()}
        xExtents={xExtents}
      >
        <Chart id={1} yExtents={d => [d.high, d.low]}>
          <XAxis axisAt="bottom" orient="bottom" ticks={6} />
          <YAxis axisAt="left" orient="left" ticks={5} />
          <CandlestickSeries width={timeIntervalBarWidth(utcDay)} />
        </Chart>
      </ChartCanvas>
    );
  }
}

