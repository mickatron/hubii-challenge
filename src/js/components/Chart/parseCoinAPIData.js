import { timeParse } from 'd3-time-format';

const parseDate = timeParse('%Y-%m-%d');

export default function parseCoinAPIData(dataArray) {
  return dataArray.map(data => ({
    date: parseDate(data.time_period_start.split('T')[0]),
    open: data.price_open,
    high: data.price_high,
    low: data.price_low,
    close: data.price_close,
    volume: data.volume_traded,
  })).reverse();
}
