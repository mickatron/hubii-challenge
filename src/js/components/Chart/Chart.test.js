import React from 'react';
import { shallow } from 'enzyme';
import Chart from './Chart';

const date = new Date('2018-04-05T00:00:00');
function createChartData(length = 9, data = []) {
  if (length === -1) {
    return data;
  }
  date.setDate(date.getDate() - 1);
  data.push({
    close: 25.710416,
    date,
    high: 25.835021381744056,
    low: 25.411360259406774,
    open: 25.436282332605284,
    volume: 38409100,
  });
  return createChartData(length - 1, data);
}

describe('Chart', () => {
  let component = null;
  describe('renders', () => {
    it('a snapshot', () => {
      component = shallow((
        <Chart
          data={createChartData()}
        />
      ));
      expect(component).toMatchSnapshot();
    });
  });
});
