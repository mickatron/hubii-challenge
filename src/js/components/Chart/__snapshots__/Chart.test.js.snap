// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CandleStickChart
    className={undefined}
    data={
        Array [
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
            Object {
              "close": 25.710416,
              "date": 2018-03-25T13:00:00.000Z,
              "high": 25.835021381744056,
              "low": 25.411360259406774,
              "open": 25.436282332605284,
              "volume": 38409100,
            },
          ]
    }
    ratio={undefined}
    width={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Chart
        flipYScale={false}
        id={1}
        onContextMenu={[Function]}
        origin={
                Array [
                        0,
                        0,
                      ]
        }
        padding={0}
        yExtents={[Function]}
        yPan={true}
        yPanEnabled={false}
        yScale={[Function]}
>
        <XAxis
                axisAt="bottom"
                className="react-stockcharts-x-axis"
                domainClassName="react-stockcharts-axis-domain"
                fill="none"
                fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
                fontSize={12}
                fontWeight={400}
                getMouseDelta={[Function]}
                innerTickSize={5}
                opacity={1}
                orient="bottom"
                outerTickSize={0}
                showDomain={true}
                showTickLabel={true}
                showTicks={true}
                stroke="#000000"
                strokeWidth={1}
                tickPadding={6}
                tickStroke="#000000"
                tickStrokeOpacity={1}
                ticks={6}
                xZoomHeight={25}
                zoomEnabled={true}
        />
        <YAxis
                axisAt="left"
                className="react-stockcharts-y-axis"
                domainClassName="react-stockcharts-axis-domain"
                fill="none"
                fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
                fontSize={12}
                fontWeight={400}
                getMouseDelta={[Function]}
                innerTickSize={5}
                opacity={1}
                orient="left"
                outerTickSize={0}
                showDomain={true}
                showTickLabel={true}
                showTicks={true}
                stroke="#FFFFFF"
                strokeWidth={1}
                tickPadding={6}
                tickStroke="#000000"
                tickStrokeOpacity={1}
                ticks={5}
                yZoomWidth={40}
                zoomEnabled={true}
        />
        <CandlestickSeries
                candleClassName="react-stockcharts-candlestick-candle"
                candleStrokeWidth={0.5}
                className="react-stockcharts-candlestick"
                classNames={[Function]}
                clip={true}
                fill={[Function]}
                opacity={0.5}
                stroke="#000000"
                wickClassName="react-stockcharts-candlestick-wick"
                wickStroke="#000000"
                width={[Function]}
                widthRatio={0.8}
                yAccessor={[Function]}
        />
</Chart>,
      "clamp": false,
      "className": "Chart",
      "data": Array [
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
        Object {
          "close": 25.710416,
          "date": 2018-03-25T13:00:00.000Z,
          "high": 25.835021381744056,
          "low": 25.411360259406774,
          "open": 25.436282332605284,
          "volume": 38409100,
        },
      ],
      "defaultFocus": true,
      "disableInteraction": false,
      "displayXAccessor": [Function],
      "flipXScale": false,
      "height": 400,
      "maintainPointsPerPixelOnResize": true,
      "margin": Object {
        "bottom": 30,
        "left": 50,
        "right": 50,
        "top": 10,
      },
      "minPointsPerPxThreshold": 0.01,
      "mouseMoveEvent": true,
      "onLoadMore": [Function],
      "onSelect": [Function],
      "padding": 0,
      "panEvent": true,
      "pointsPerPxThreshold": 2,
      "postCalculator": [Function],
      "ratio": undefined,
      "seriesName": "HubiiChallenge",
      "type": "svg",
      "useCrossHairStyleCursor": true,
      "width": undefined,
      "xAccessor": [Function],
      "xExtents": Array [
        2018-03-25T13:00:00.000Z,
        undefined,
      ],
      "xScale": [Function],
      "zIndex": 1,
      "zoomAnchor": [Function],
      "zoomEvent": true,
      "zoomMultiplier": 1.1,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <XAxis
            axisAt="bottom"
            className="react-stockcharts-x-axis"
            domainClassName="react-stockcharts-axis-domain"
            fill="none"
            fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
            fontSize={12}
            fontWeight={400}
            getMouseDelta={[Function]}
            innerTickSize={5}
            opacity={1}
            orient="bottom"
            outerTickSize={0}
            showDomain={true}
            showTickLabel={true}
            showTicks={true}
            stroke="#000000"
            strokeWidth={1}
            tickPadding={6}
            tickStroke="#000000"
            tickStrokeOpacity={1}
            ticks={6}
            xZoomHeight={25}
            zoomEnabled={true}
/>,
          <YAxis
            axisAt="left"
            className="react-stockcharts-y-axis"
            domainClassName="react-stockcharts-axis-domain"
            fill="none"
            fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
            fontSize={12}
            fontWeight={400}
            getMouseDelta={[Function]}
            innerTickSize={5}
            opacity={1}
            orient="left"
            outerTickSize={0}
            showDomain={true}
            showTickLabel={true}
            showTicks={true}
            stroke="#FFFFFF"
            strokeWidth={1}
            tickPadding={6}
            tickStroke="#000000"
            tickStrokeOpacity={1}
            ticks={5}
            yZoomWidth={40}
            zoomEnabled={true}
/>,
          <CandlestickSeries
            candleClassName="react-stockcharts-candlestick-candle"
            candleStrokeWidth={0.5}
            className="react-stockcharts-candlestick"
            classNames={[Function]}
            clip={true}
            fill={[Function]}
            opacity={0.5}
            stroke="#000000"
            wickClassName="react-stockcharts-candlestick-wick"
            wickStroke="#000000"
            width={[Function]}
            widthRatio={0.8}
            yAccessor={[Function]}
/>,
        ],
        "flipYScale": false,
        "id": 1,
        "onContextMenu": [Function],
        "origin": Array [
          0,
          0,
        ],
        "padding": 0,
        "yExtents": [Function],
        "yPan": true,
        "yPanEnabled": false,
        "yScale": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "axisAt": "bottom",
            "className": "react-stockcharts-x-axis",
            "domainClassName": "react-stockcharts-axis-domain",
            "fill": "none",
            "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 400,
            "getMouseDelta": [Function],
            "innerTickSize": 5,
            "opacity": 1,
            "orient": "bottom",
            "outerTickSize": 0,
            "showDomain": true,
            "showTickLabel": true,
            "showTicks": true,
            "stroke": "#000000",
            "strokeWidth": 1,
            "tickPadding": 6,
            "tickStroke": "#000000",
            "tickStrokeOpacity": 1,
            "ticks": 6,
            "xZoomHeight": 25,
            "zoomEnabled": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "axisAt": "left",
            "className": "react-stockcharts-y-axis",
            "domainClassName": "react-stockcharts-axis-domain",
            "fill": "none",
            "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 400,
            "getMouseDelta": [Function],
            "innerTickSize": 5,
            "opacity": 1,
            "orient": "left",
            "outerTickSize": 0,
            "showDomain": true,
            "showTickLabel": true,
            "showTicks": true,
            "stroke": "#FFFFFF",
            "strokeWidth": 1,
            "tickPadding": 6,
            "tickStroke": "#000000",
            "tickStrokeOpacity": 1,
            "ticks": 5,
            "yZoomWidth": 40,
            "zoomEnabled": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "candleClassName": "react-stockcharts-candlestick-candle",
            "candleStrokeWidth": 0.5,
            "className": "react-stockcharts-candlestick",
            "classNames": [Function],
            "clip": true,
            "fill": [Function],
            "opacity": 0.5,
            "stroke": "#000000",
            "wickClassName": "react-stockcharts-candlestick-wick",
            "wickStroke": "#000000",
            "width": [Function],
            "widthRatio": 0.8,
            "yAccessor": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Chart
          flipYScale={false}
          id={1}
          onContextMenu={[Function]}
          origin={
                    Array [
                              0,
                              0,
                            ]
          }
          padding={0}
          yExtents={[Function]}
          yPan={true}
          yPanEnabled={false}
          yScale={[Function]}
>
          <XAxis
                    axisAt="bottom"
                    className="react-stockcharts-x-axis"
                    domainClassName="react-stockcharts-axis-domain"
                    fill="none"
                    fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
                    fontSize={12}
                    fontWeight={400}
                    getMouseDelta={[Function]}
                    innerTickSize={5}
                    opacity={1}
                    orient="bottom"
                    outerTickSize={0}
                    showDomain={true}
                    showTickLabel={true}
                    showTicks={true}
                    stroke="#000000"
                    strokeWidth={1}
                    tickPadding={6}
                    tickStroke="#000000"
                    tickStrokeOpacity={1}
                    ticks={6}
                    xZoomHeight={25}
                    zoomEnabled={true}
          />
          <YAxis
                    axisAt="left"
                    className="react-stockcharts-y-axis"
                    domainClassName="react-stockcharts-axis-domain"
                    fill="none"
                    fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
                    fontSize={12}
                    fontWeight={400}
                    getMouseDelta={[Function]}
                    innerTickSize={5}
                    opacity={1}
                    orient="left"
                    outerTickSize={0}
                    showDomain={true}
                    showTickLabel={true}
                    showTicks={true}
                    stroke="#FFFFFF"
                    strokeWidth={1}
                    tickPadding={6}
                    tickStroke="#000000"
                    tickStrokeOpacity={1}
                    ticks={5}
                    yZoomWidth={40}
                    zoomEnabled={true}
          />
          <CandlestickSeries
                    candleClassName="react-stockcharts-candlestick-candle"
                    candleStrokeWidth={0.5}
                    className="react-stockcharts-candlestick"
                    classNames={[Function]}
                    clip={true}
                    fill={[Function]}
                    opacity={0.5}
                    stroke="#000000"
                    wickClassName="react-stockcharts-candlestick-wick"
                    wickStroke="#000000"
                    width={[Function]}
                    widthRatio={0.8}
                    yAccessor={[Function]}
          />
</Chart>,
        "clamp": false,
        "className": "Chart",
        "data": Array [
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
          Object {
            "close": 25.710416,
            "date": 2018-03-25T13:00:00.000Z,
            "high": 25.835021381744056,
            "low": 25.411360259406774,
            "open": 25.436282332605284,
            "volume": 38409100,
          },
        ],
        "defaultFocus": true,
        "disableInteraction": false,
        "displayXAccessor": [Function],
        "flipXScale": false,
        "height": 400,
        "maintainPointsPerPixelOnResize": true,
        "margin": Object {
          "bottom": 30,
          "left": 50,
          "right": 50,
          "top": 10,
        },
        "minPointsPerPxThreshold": 0.01,
        "mouseMoveEvent": true,
        "onLoadMore": [Function],
        "onSelect": [Function],
        "padding": 0,
        "panEvent": true,
        "pointsPerPxThreshold": 2,
        "postCalculator": [Function],
        "ratio": undefined,
        "seriesName": "HubiiChallenge",
        "type": "svg",
        "useCrossHairStyleCursor": true,
        "width": undefined,
        "xAccessor": [Function],
        "xExtents": Array [
          2018-03-25T13:00:00.000Z,
          undefined,
        ],
        "xScale": [Function],
        "zIndex": 1,
        "zoomAnchor": [Function],
        "zoomEvent": true,
        "zoomMultiplier": 1.1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <XAxis
              axisAt="bottom"
              className="react-stockcharts-x-axis"
              domainClassName="react-stockcharts-axis-domain"
              fill="none"
              fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
              fontSize={12}
              fontWeight={400}
              getMouseDelta={[Function]}
              innerTickSize={5}
              opacity={1}
              orient="bottom"
              outerTickSize={0}
              showDomain={true}
              showTickLabel={true}
              showTicks={true}
              stroke="#000000"
              strokeWidth={1}
              tickPadding={6}
              tickStroke="#000000"
              tickStrokeOpacity={1}
              ticks={6}
              xZoomHeight={25}
              zoomEnabled={true}
/>,
            <YAxis
              axisAt="left"
              className="react-stockcharts-y-axis"
              domainClassName="react-stockcharts-axis-domain"
              fill="none"
              fontFamily="Helvetica Neue, Helvetica, Arial, sans-serif"
              fontSize={12}
              fontWeight={400}
              getMouseDelta={[Function]}
              innerTickSize={5}
              opacity={1}
              orient="left"
              outerTickSize={0}
              showDomain={true}
              showTickLabel={true}
              showTicks={true}
              stroke="#FFFFFF"
              strokeWidth={1}
              tickPadding={6}
              tickStroke="#000000"
              tickStrokeOpacity={1}
              ticks={5}
              yZoomWidth={40}
              zoomEnabled={true}
/>,
            <CandlestickSeries
              candleClassName="react-stockcharts-candlestick-candle"
              candleStrokeWidth={0.5}
              className="react-stockcharts-candlestick"
              classNames={[Function]}
              clip={true}
              fill={[Function]}
              opacity={0.5}
              stroke="#000000"
              wickClassName="react-stockcharts-candlestick-wick"
              wickStroke="#000000"
              width={[Function]}
              widthRatio={0.8}
              yAccessor={[Function]}
/>,
          ],
          "flipYScale": false,
          "id": 1,
          "onContextMenu": [Function],
          "origin": Array [
            0,
            0,
          ],
          "padding": 0,
          "yExtents": [Function],
          "yPan": true,
          "yPanEnabled": false,
          "yScale": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "axisAt": "bottom",
              "className": "react-stockcharts-x-axis",
              "domainClassName": "react-stockcharts-axis-domain",
              "fill": "none",
              "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "fontWeight": 400,
              "getMouseDelta": [Function],
              "innerTickSize": 5,
              "opacity": 1,
              "orient": "bottom",
              "outerTickSize": 0,
              "showDomain": true,
              "showTickLabel": true,
              "showTicks": true,
              "stroke": "#000000",
              "strokeWidth": 1,
              "tickPadding": 6,
              "tickStroke": "#000000",
              "tickStrokeOpacity": 1,
              "ticks": 6,
              "xZoomHeight": 25,
              "zoomEnabled": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "axisAt": "left",
              "className": "react-stockcharts-y-axis",
              "domainClassName": "react-stockcharts-axis-domain",
              "fill": "none",
              "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "fontWeight": 400,
              "getMouseDelta": [Function],
              "innerTickSize": 5,
              "opacity": 1,
              "orient": "left",
              "outerTickSize": 0,
              "showDomain": true,
              "showTickLabel": true,
              "showTicks": true,
              "stroke": "#FFFFFF",
              "strokeWidth": 1,
              "tickPadding": 6,
              "tickStroke": "#000000",
              "tickStrokeOpacity": 1,
              "ticks": 5,
              "yZoomWidth": 40,
              "zoomEnabled": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "candleClassName": "react-stockcharts-candlestick-candle",
              "candleStrokeWidth": 0.5,
              "className": "react-stockcharts-candlestick",
              "classNames": [Function],
              "clip": true,
              "fill": [Function],
              "opacity": 0.5,
              "stroke": "#000000",
              "wickClassName": "react-stockcharts-candlestick-wick",
              "wickStroke": "#000000",
              "width": [Function],
              "widthRatio": 0.8,
              "yAccessor": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
